packages/headless/src/controllers/facets/facet-search/facet-search.ts

```diff
@@ -77,6 +80,17 @@ export function buildGenericFacetSearch<T extends FacetSearchState>(
       dispatch(executeSearch(logFacetSelect({facetId, facetValue})));
     },
 
+    /** Resets the query and empties the values. */
+    clear() {
+      dispatch(
+        updateFacetSearch({
+          facetId,
+          query: defaultFacetSearchOptions.query,
+        })
+      );
+      dispatch(clearFacetSearchResults({facetId}));
```

Comment:
Instead of separating the behaviour across two actions, why not combine them into a single, cohesive action? e.g.

```
clearFacetSearch({facetId})
```

From the the [last thread](https://github.com/coveo/ui-kit/pull/736#discussion_r619358778), I don't think there is a use-case for someone to clear just the facet search results without resulting in a worse experience?

Reply:
> If on blur, someone wants to reset the facet search query and results, then they can call the new controller method you are adding. However, the action powering the interaction should also be cohesive so that developers are not dependent on the controller.

From the last thread, I made the action able to be used on its own, without having to change the numberOfValues with updateFacetSearch. Actions are usually granular though, and give more power to the users, for the search box for example, we have separate actions for clearing results or clearing results + query.

I can merge them together though, it doesn't really change anything to our use case, it's more arguing about potential customer use cases.